modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractKotlinCodegen.java

```diff
@@ -402,10 +402,10 @@ public String toEnumVarName(String value, String datatype) {
         if (reservedWords.contains(modified)) {
             return escapeReservedWord(modified);
         }
-
-        return modified;
+        return sanitizeKotlinSpecificNames(modified);
```

Comment:
Can you add a comment here about why this is necessary, considering this is called at the start of the method? Otherwise I think it would be confusing to maintainers.

Reading it, I see that the issue is an enum input of `1enum` becomes `_1enum` at the start of the method, then `camelCase` (the default) and `PascalCase` end up removing that beginning underscore, resulting in the invalid `1enum`. I had to look at it for a second to understand the need at the end of the method, but I'm not sure all contributors would pick up on the nuance even with the comments in the case statements.